### ISSUE TITLE
get_posts returns only 20 posts when logged in

### ISSUE THREAD

**Issue by BodomBeach** [kevinzg/facebook-scraper#532]
I am trying to scrap group=120778514747417 which is a public group.

`print(len(list(get_posts(group=120778514747417))))`

If I don't run `set_cookies` first, the scraper works fine but I get banned pretty quickly, probably because I am unlogged.
If I run `set_cookies` first with my account which is part of this group, scraper only returns 20 posts and stops there.
Logger says the following at the end:
`No raw posts (<article> elements) were found in this page.`
`Page parser did not find next page URL`
`20`

How can I scrap more posts being logged in ?
If not possible, what is the recommended throttling to scrap without cookies ?
Thanks and great work btw!

**Comment by BodomBeach** [comment]
hi @neon-ninja 
any update on this issue ? this is making group scraping impossible.

I am now considering to reverse-engineer the www version of facebook to scrape groups. 
Would you be interested in my results if I manage to achieve something ?
Also, have you already fiddled around the www version ? maybe you could give me a headstart

**Comment by neon-ninja** [comment]
Hi @BodomBeach,
No update, this scraper still doesn't support the desktop version of Facebook. Yes, I would accept a pull request implementing this feature.

A bit - I think it would be best to use selenium, chromium/chrome headless, webdriver-manager (https://pypi.org/project/webdriver-manager/) and target the accessibility selectors (aria-*) in the Facebook desktop. The accessibility tab in Chrome dev tools would be useful for figuring out selectors. Essentially you'd be building an automated screen reader.

**Comment by chribell** [comment]
It seems that facebook have changed the next link format. By adding a new regex in the PageParser class, I've managed to fetch the next pages. Moreover, 
```
class PageParser:
    """Class for Parsing a single page on a Page"""
    ....
    cursor_regex_5 = re.compile(r'href:"/groups/(.*?)multi_permalinks"')
    ....

    def get_next_page(self) -> Optional[URL]:
        ....
        match = self.cursor_regex_5.search(self.cursor_blob)
        if match:
            return match.groups()[0]
        return None

```

Maybe giving the regex as an argument is a more robust solution (as long as fb gives the complete href in the page and doesn't suddenly construct it on the fly)?

**Comment by neon-ninja** [comment]
Sounds like a good fix, please submit a pull request

**Comment by BodomBeach** [comment]
It looks indeed like the href for next page. However, visiting this next page url loads the group page with zero article in it. Tried on several groups.
Have you managed to load more results ? @chribell

**Comment by chribell** [comment]
Yes, I can fetch more results until I get temp banned. In my scenario, the logged in user is the admin of a private group.

**Comment by neon-ninja** [comment]
Oh, I've just realised - this kind of group pagination URL is already handled by this regex - `href[=:]"(\/groups\/[^"]+bac=[^"]+)"`. All group pagination should be handled by the GroupPageParser class - given you've put it in PageParser, suggests you're not calling get_posts correctly @chribell. For a group, call `get_posts` like this: `get_posts(group="group_name")`

**Comment by chribell** [comment]
You are right @neon-ninja. I didn't pass the group id as a named argument. If I do it works as intended. Sorry for the confusion, my bad.

**Comment by pietro-ayy** [comment]
Hi, has anyone been able to fix the issue? Thanks

**Comment by neon-ninja** [comment]
This is a bug in Facebook itself, there's not a lot we can do about it

**Comment by annaantoniou** [comment]
Any updates on this?
